include/mitsuba/render/scene.h

```diff
@@ -79,6 +79,73 @@ class MTS_EXPORT_RENDER Scene : public Object {
     //! @{ \name Sampling interface
     // =============================================================
 
+    /**
+     * \brief Sample one emitter in the scene.
+     * If possible, it is sampled proportional to its radiance.
+     *
+     * \param sample
+     *    A uniformly distributed number in [0, 1).
+     *
+     * \return
+     *    The index of the chosen emitter along with the sampling weight (equal
+     *    to the inverse PDF).
+     */
+    std::pair<UInt32, Float> sample_emitter(Float index_sample,
+                                            Mask active = true) const;
+
+    /**
+     * \brief Sample one emitter in the scene and rescale the random sample for reuse.
+     * If possible, it is sampled proportional to its radiance.
+     *
+     * \param sample
+     *    A uniformly distributed number in [0, 1).
+     *
+     * \return
+     *    The index of the chosen emitter along with the sampling weight (equal
+     *    to the inverse PDF), and the rescaled random sample.
+     */
+    std::tuple<UInt32, Float, Float>
+    sample_emitter_reuse(Float index_sample, Mask active = true) const;
```

Comment:
Could this method take the random number as reference and update it "in-place"? In which case we wouldn't both methods actually. Or is there a case where we explicitly want both?

Reply:
Kept only `sample_emitter_reuse` but renamed it `sample_emitter`, indeed we don't really need both.